### PR TITLE
Create new game, list games and join

### PR DESCRIPTION
**Pull request by mattschmitter**
This was initially started in the create_new_game branch, but with the last merges I created this new branch to deal with the conflicts.  There are a couple of new things we may want to address before merging this:
- I removed line for the games resources from routes.rb because I used update for the join method.  :except => [:update, :patch]
- In the games controller, I was thinking we may want to authenticate the actions (before_action :authenticate_user!) but I'm not sure how that interacts with (skip_before_action :verify_authenticity_token).
- I also just took an initial stab at the games show page.  Currently the view does the work in showing which games you're a part of and those that are open.  This logic could reside elsewhere, though.
- My tests are very weak

### CODE REVIEW

**Comment by tbowyer** [comment]
I forgot to tell everyone that eventually I removed the the skip_before_action :verify_authenticity_token but I haven't submitted a pull request since doing so.  I had that in there originally because I couldn't figure out how to get the AJAX to work without adding it to the top of the controller.  After doing some research I found that it's related to the CSRF tag in our header on application.html.erb and now in my AJAX code I have this which allows for protection against CSRF:

```
                        beforeSend: function(xhr) {xhr.setRequestHeader('X-CSRF-Token', $('meta[name="csrf-token"]').attr('content'))},
```

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -1,10 +1,24 @@
 class GamesController < ApplicationController
   skip_before_action :verify_authenticity_token
+  
+  def create
+    Game.create( :white_player_id => current_user.id )
+    flash[:notice] = "Your game has been created.  While you wait for an opponent, consider joining another open game."
```
Comment (ilyakrasnov) [inline app/controllers/games_controller.rb:6]: :+1:

**Comment by ilyakrasnov** [comment]
:+1: anyone else a thumbsup? Can be merged

**Comment by ghost** [comment]
:+1:

**Pull request merged**